src/Servers/Kestrel/shared/HPackHeaderWriter.cs

```diff
@@ -35,20 +49,20 @@ public static bool BeginEncodeHeaders(int statusCode, DynamicHPackEncoder hpackE
 
         if (!headersEnumerator.MoveNext())
         {
-            return true;
+            return HeaderWriteResult.Done;
         }
 
-        // We're ok with not throwing if no headers were encoded because we've already encoded the status.
+        // We're ok with not increasing the buffer size if no headers were encoded because we've already encoded the status.
```

Comment:
```suggestion
        // Since we've already encoded the status, we know we didn't start with an empty buffer.  We don't need to increase it immediately because
```

Reply:
applying in a follow-up commit